phalcon/mailer.zep

```diff
@@ -0,0 +1,624 @@
+    * $objMail->msg('mensaje de prueba');
+    * $objMail->addAttachment('/file.odt');
+    *
+    * // Add addressee for to
+    * $objMail->addTo('prueba_to@address.com', 'pruba_name_to');
+    *
+    * // Send mail
+    * $objMail->send();
+    * </code>
+    *
+    **/
+
+class Mailer {
+    protected server {
+        set, get
+    };
+    protected port {
+        set, get
+    };
+    protected username {
+        set, get
+    };
+    protected password {
+        set, get
+    };
+    protected secure{
+        set, get
+    };
+    
+    protected smtpAuth = false {
+        set, get
+    };
+    protected xMailer = "Phalcon Mailer" {
+        set, get
+    };
+    protected priority = 3 {
+        set, get
+    };
+    
+    private from = "";
+    private fromName = "";
+    private replyTo = "";
+    private replyToName = "";
+    private recipients = [];
+    
+    protected subject = "" {
+        set, get
+    };
+    protected message = "" {
+        set, get
+    };
+    
+    protected charset = "\"utf-8\"" { 
+        set, get
+    };
+    
+    /**
+     * Gett and Sett contentType
+     * Can be set to: text/plain, text/html, multipart/mixed
+     *
+     * @param string 
+     * 
+     */
+    protected contentType = "multipart/alternative" { 
+        set, get
+    };
+    
+    protected transferEncodeing = "8bit" { 
+        set, get
+    }; 
+    protected altBody = "" { 
+        set, get
+    };
+    protected isLogin = false { 
+        set, get
+    };
+    protected cc = []{ 
+        set, get
+    };
+    protected bcc = [] { 
+        set, get
+    };
+    protected attachments = [] { 
+        set, get
+    };
+
+    private conn;
+    private newline = "\r\n";
+    private localhost = "localhost";
+    private timeout = "15";
+    private debug = false;
+    
+    /**
+     * Constructor
+     */
+    public function __construct(var parameters){
+        this->setParams(parameters);
+    }
+    
+    /**
+     * Set parameters
+     *
+     * @param array   parameters
+     * 
+     */
+    public function setParams(var parameters) -> void {
+        var key, value;
+        
+        if (typeof parameters != "array") {
+            throw new Exception("Mail error: Parameters not array");
+        }
+        else {
+            for key, value in parameters {
+                if property_exists(this,key) {
+                    this->setParam(key, value);
+                }
+            }
+        }
+    }
+
+    /**
+     * Set single parameter
+     *
+     * @param string  key
+     * @param type_key  value
+     * 
+     */
+    public function setParam(var key, var value) -> void {
+        let this->{key} = value;
+    }
+    
+
+    /**
+     * Connect to the server
+     * return status conection
+     * 
+     */
+    private function connect() -> boolean {
+        if strtolower(trim(this->secure)) == "ssl" {
+            let this->server = "ssl://" . this->server;
```

Comment:
Not sure but it's better to store any value of this->property or this->method() in zephir to variable before using it, especially multiple times.